app/routes/exercises.rb

```diff
@@ -66,6 +66,28 @@ class Exercises < Core
         redirect '/dashboard'
       end
 
+      post '/exercises/:key/request-for-help' do |key|
+        exercise = UserExercise.find_by_key(key)
+        unless current_user.owns?(exercise)
+          flash[:notice] = "Only the author may request help."
+          redirect '/'
+        end
+        exercise.request_for_help!(current_user)
+        flash[:success] = "You have requested for help on this submission"
```

Comment:
You have requested ~~for~~ help on this submission.

I'd also like to end with a period.